nameserver/nameserver.go

```diff
@@ -83,12 +176,50 @@ func (n *Nameserver) broadcastEntries(es ...Entry) {
 	})
 }
 
-func (n *Nameserver) AddEntry(hostname, containerid string, origin mesh.PeerName, addr address.Address) {
+func (n *Nameserver) AddEntry(hostname, containerid string,
+	origin mesh.PeerName, addr address.Address) {
+
+	var entries Entries
+	var found bool
+
+	n.Lock()
+	// Restore the stopped entries first
```

Comment:
Good find. I haven't addressed such situation, so currently, it will restore entries with a previous IP address being set, which is wrong.